app/controllers/search.controller.js

```diff
@@ -1,55 +1,136 @@
+const cron = require('node-schedule');
+const puppeteer = require('puppeteer-extra');
+const { Op } = require('sequelize');
+const StealthPlugin = require('puppeteer-extra-plugin-stealth');
+
 const db = require('../models');
+const KeywordManager = require('../event-emitters/keywordsManager');
+const timer = require('../utils/timer');
 
-const puppeteer = require('puppeteer');
-const { Op } = require('sequelize');
+// Bypass mass-searching Google (sometimes it works, and sometimes it's not)
+puppeteer.use(StealthPlugin());
 
 const Keyword = db.keyword;
 
-exports.searchByEachKeyWords = async (req, res) => {
-  const { keywords } = req.body;
+const keywordManager = new KeywordManager();
 
+keywordManager.on('crawl', async (userId, keywords) => {
```

Comment:
Using event emitters falls short of the expectation as a background process queue (e.g., using https://optimalbits.github.io/bull/) should be used. Frameworks like NestJS provide examples on how to do: https://docs.nestjs.com/techniques/queues

Reply:
There is a lot of ways to implement the background process. So that's why I chose the event emitters. Thanks a lot for your suggest. I will try that!